homeassistant/components/orangepi_gpio/cover.py

```diff
@@ -0,0 +1,115 @@
+"""Support for controlling a Orange Pi cover."""
+import logging
+from time import sleep
+
+import voluptuous as vol
+
+from homeassistant.components.cover import CoverDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.components import orangepi_gpio
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COVERS = 'covers'
+CONF_PINMODE = 'pinmode'
+CONF_RELAY_PIN = 'relay_pin'
+CONF_RELAY_TIME = 'relay_time'
+CONF_STATE_PIN = 'state_pin'
+CONF_INVERT_STATE = 'invert_state'
+CONF_INVERT_RELAY = 'invert_relay'
+
+PINMODES = ['pc', 'zeroplus', 'zeroplus2', 'deo', 'neocore2']
+
+DEFAULT_RELAY_TIME = .2
+DEFAULT_INVERT_STATE = False
+DEFAULT_INVERT_RELAY = False
+DEPENDENCIES = ['orangepi_gpio']
+
+_COVERS_SCHEMA = vol.All(
+    cv.ensure_list,
+    [
+        vol.Schema({
+            CONF_NAME: cv.string,
+            CONF_RELAY_PIN: cv.positive_int,
+            CONF_STATE_PIN: cv.positive_int,
+        })
+    ]
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_COVERS): _COVERS_SCHEMA,
+    vol.Required(CONF_PINMODE): vol.In(PINMODES),
+    vol.Optional(CONF_RELAY_TIME, default=DEFAULT_RELAY_TIME): cv.positive_int,
+    vol.Optional(CONF_INVERT_STATE, default=DEFAULT_INVERT_STATE): cv.boolean,
+    vol.Optional(CONF_INVERT_RELAY, default=DEFAULT_INVERT_RELAY): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Orange Pi cover platform."""
+    pinmode = config.get('pinmode')
```

Comment:
Use the constants defined
```suggestion
    pinmode = config[CONF_PINMODE]
```